lib/compute/jenkins-main-node.ts

```diff
@@ -365,7 +365,7 @@ export class JenkinsMainNode {
       InitCommand.shellCommand('/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -c file:/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json -s'),
 
       InitCommand.shellCommand(dataRetentionProps.dataRetention
-        ? `mkdir /var/lib/jenkins && mount -t efs ${efsId} /var/lib/jenkins`
+        ? `mkdir /var/lib/jenkins && mount -t efs ${efsId} /var/lib/jenkins && rm -rf /var/lib/jenkins/caches/*`
```

Comment:
@peterzhuamazon is it safe to delete all caches or only tool specific caches ?

Reply:
My point is usually the tool renaming is not done very often, so its not required to remove the all the caches

Reply:
fixed now to only remove jdks.

Reply:
@peterzhuamazon  we dont have to touch any caches folder, let the caches be there, only issue we see is because we have changed the tool names and that messed up.